### PR TITLE
Bumping versions to 3.0.0

### PR DESCRIPTION
**Pull request by nsillik**
### Fixed

- Fixed an issue where `AttributedLabel` would not properly handle tapping on links when a label was stretched.

### Added

- `AccessibilityElement` now supports providing arbitrary strings to assistive devices using the `AXCustomContent` protocol. 

### Changed

- The behavior of `name` of `ElementPreview` has been change, affecting the SwiftUI `previewName`. Instead of including device or size information (i.e. `sizeThatFits - \(name)`), it now either defaults to the Xcode default if given an empty string, and shows _only_ the `name` if `name` is non-empty.
- Updated minimum deployment target from iOS 14 to iOS 15.

### Internal
- Updated CI to use M1 machines, Xcode 15.1, and Ruby 3.2.2.
- Added iOS 17 snapshot images.
- Bump Swift version to 5.9.
- Update Ruby gems.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -120,6 +120,7 @@ GEM
 
 PLATFORMS
   arm64-darwin-22
+  arm64-darwin-23
```
Comment (n8chur) [inline Gemfile.lock:123]: Was this change intentional? It seems fine, but wanted to double check.

**Review by n8chur — APPROVED** [review]

File: Gemfile.lock (hunk 1 above)
Comment (nsillik) [reply Gemfile.lock:123]: This is just Sonoma, figured I'd keep it in otherwise it's just gonna keep happening
